src/App.tsx

```diff
@@ -14,11 +15,114 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  isListVisible: boolean;
+  listOfGoods: string[];
+};
+
+class App extends React.PureComponent<{}, State> {
+  state: State = {
+    isListVisible: false,
+    listOfGoods: goodsFromServer,
+  };
+
+  reverseList = () => {
+    this.setState(state => ({
+      listOfGoods: [...state.listOfGoods].reverse(),
+    }));
+  };
+
+  sortAlphabetically = () => {
+    this.setState(state => ({
+      listOfGoods: [...state.listOfGoods].sort((a, b) => a.localeCompare(b)),
+    }));
+  };
+
+  sortbyLength = () => {
+    this.setState(state => ({
+      listOfGoods: [...state.listOfGoods].sort((a, b) => a.length - b.length),
+    }));
+  };
+
+  reset = () => {
```

Comment:
Try to add the word "Handler" to the onClick method name of each button. This is the best practice. For example, "resetHandler".